packages/eslint-plugin/src/utils/configHelpers.js

```diff
@@ -132,18 +132,31 @@ module.exports = {
     }
 
     /**
-       * Note that this way of reading tsconfig does not account for extends, but that's okay since
-       * typically include will not be specified that way
+       * Note that this approach only accounts for a single level of extends. JJU is used for parsing
+       * the tsconfig because Typescript functions are more complex than necessary.
        *
        * @type {{
-          extends: string[];
+          extends: string;
           include: string[];
-          excludes: string[];
+          exclude: string[];
           compilerOptions: Record<string,unknown>;
           references?: Array<{path:string}>
           }}
        */
-    const tsconfig = jju.parse(fs.readFileSync(tsconfigPath).toString());
+    let tsconfig = jju.parse(fs.readFileSync(tsconfigPath).toString());
+
+    /**
+     * Handle any necessary extends merging here, make sure to treat just like native tsconfigs would
+     */
+    if (tsconfig.extends) {
+      const parentTsConfigPath = path.join(path.dirname(tsconfigPath), tsconfig.extends);
+      /** @type { typeof tsconfig } */
+      const parentTsConfig = jju.parse(fs.readFileSync(parentTsConfigPath).toString());
+
+      // Extending config overrides parent files, include and exclue
+      // https://www.typescriptlang.org/tsconfig#extends
+      tsconfig = { ...parentTsConfig, ...tsconfig };
```

Comment:
You need to merge the compiler options too
```suggestion
      tsconfig = { ...parentTsConfig, ...tsconfig, compilerOptions: { ...parentTsConfig.compilerOptions, ...tsConfig.compilerOptions };
```